lib/src/client/dio_interceptor.dart

```diff
@@ -41,57 +44,64 @@ class RequestsSignatureInterceptor extends Interceptor {
   Future onRequest(
       RequestOptions options, RequestInterceptorHandler handler) async {
     try {
-      Dio dio = Dio();
-      _validateOptions();
-
-      // Sign the request before it is sent
+      log('[✓] onRequest triggered');
       await _signRequest(options);
 
-      // Proceed with the request
-      Response<dynamic> response = await dio.request(
-        options.path,
-        data: options.data,
-        options: Options(
-          method: options.method,
-          headers: options.headers,
-        ),
-      );
-
-      // Check for clock skew only if auto-retry is enabled
+      return handler.next(options);
+    } catch (err, stackTrace) {
+      log('[x] onrequest error');
+      throw RequestsSignatureException('$err\n$stackTrace');
+    }
+  }
+
+  @override
+  void onResponse(Response response, ResponseInterceptorHandler handler) {
+    _validateOptions();
+
+    try {
+      log('[✓] Initial response status code : ${response.statusCode} OK');
+
       if (!_options.disableAutoRetryOnClockSkew &&
-          (response.statusCode == HttpStatus.unauthorized ||
-              response.statusCode == HttpStatus.forbidden) &&
+          ((response.statusCode == HttpStatus.unauthorized) ||
+              (response.statusCode == HttpStatus.forbidden)) &&
           response.headers.value(HttpHeaders.dateHeader) != null) {
-        print("HttpHeaders.dateHeader ${HttpHeaders.dateHeader}");
-        // Parse the server's timestamp from the response headers
-        final serverTimestamp =
-            DateTime.parse(response.headers.value(HttpHeaders.dateHeader)!);
-        final now = DateTime.now();
-        final diff = (now.difference(serverTimestamp).inSeconds);
-        if (diff > _options.clockSkew.inSeconds) {
-          _clockSkew = serverTimestamp.difference(now).inSeconds;
-
+        log(response.headers.value(HttpHeaders.dateHeader.toString()) == null
+            ? '[!] HttpHeaders.dateHeader NULL'
+            : '[✓] HttpHeaders.dateHeader NOT NULL');
+        final serverDate =
+            int.tryParse(response.headers.value(HttpHeaders.dateHeader)!);
+        final now = _getTime(
+            DateTime.tryParse(response.headers.value(HttpHeaders.dateHeader)!));
+
+        if (((serverDate! - now).abs()) > _options.clockSkew.inSeconds) {
+          log('[✓] onResponse triggered [IF_BLOCK]');
```

Comment:
Can't you specify a log level? Based on [this](https://dev.azure.com/nventive/Practice%20committees/_wiki/wikis/Committees.wiki/13733/Logging) documentation.